### PR TITLE
DepthTexture: Update depth + stencil textures to work with multi sample RTT, make formats consistent

### PR DESCRIPTION
**Pull request by gkjohnson**
Related issue: --

**Description**

I expect this hasn't been touched much since WebGL has been removed but there were a few issues I noticed with DepthTextures when I was experimenting with them for #28423:

- A stencil attachment could not be used with a 32F depth type.
- A stencil attachment could not be used when a multi sample render target is used.
- Using a stencil format didn't work at all unless using UnsignedInt248 type

You can see in the current [depth texture example](https://threejs.org/examples/?q=depth#webgl_depth_texture) that using a stencil buffer breaks the page. With this PR:

- DepthTexture.type is allowed to be set to FloatType, UnsignedIntType, and UnsignedShortType. If a stencil attachment is required then the appropriate internal type is selected. Stencil is unsupported for UnsignedShortType. The UnsignedInt248Type remains for backwards compatibility and does the same as UnsignedIntType.
- The internal depth attachment for the multi sample textures is now created with the same format as the assigned depth texture.
- Code redundancy has been reduced in WebGLTextures.
- The depth texture demo has been modified to test all the new options including render target samples option.

### CODE REVIEW

**Comment by Mugen87** [comment]
> I expect this hasn't been touched much since WebGL has been removed but there were a few issues I noticed with DepthTextures when I was experimenting with them for https://github.com/mrdoob/three.js/pull/28423:

I think this happens because of #27900. Can we set `stencil: true` in the depth texture demo and add the stencil format back to options?

File: src/renderers/webgl/WebGLTextures.js [hunk 1]
```diff
@@ -1433,74 +1459,37 @@ function WebGLTextures( _gl, extensions, state, properties, capabilities, utils,
...
 	// Setup storage for internal depth/stencil buffers and bind to correct framebuffer
 	function setupRenderBufferStorage( renderbuffer, renderTarget, isMultisample ) {
 
 		_gl.bindRenderbuffer( _gl.RENDERBUFFER, renderbuffer );
 
-		if ( renderTarget.depthBuffer && ! renderTarget.stencilBuffer ) {
-
-			let glInternalFormat = _gl.DEPTH_COMPONENT24;
-
-			if ( isMultisample || useMultisampledRTT( renderTarget ) ) {
```
Comment (Mugen87) [inline src/renderers/webgl/WebGLTextures.js:1446]: I'm not sure why the if checks are different in the `renderTarget.depthBuffer && ! renderTarget.stencilBuffer` branch compared to `renderTarget.depthBuffer && renderTarget.stencilBuffer`. Why did you choose the below pattern?
```js
if ( isUseMultisampledRTT ) {

	multisampledRTTExt.renderbufferStorageMultisampleEXT( _gl.RENDERBUFFER, samples, glInternalFormat, renderTarget.width, renderTarget.height );

} else if ( isMultisample ) {

	_gl.renderbufferStorageMultisample( _gl.RENDERBUFFER, samples, glInternalFormat, renderTarget.width, renderTarget.height );

} else {

	_gl.renderbufferStorage( _gl.RENDERBUFFER, glInternalFormat, renderTarget.width, renderTarget.height );

}
```
and not this one:
```js
if ( isMultisample || useMultisampledRTT( renderTarget ) ) {

	if ( useMultisampledRTT( renderTarget ) ) {

		multisampledRTTExt.renderbufferStorageMultisampleEXT( _gl.RENDERBUFFER, samples, glInternalFormat, renderTarget.width, renderTarget.height );

	} else {

		_gl.renderbufferStorageMultisample( _gl.RENDERBUFFER, samples, glInternalFormat, renderTarget.width, renderTarget.height );

	}

} else {

	_gl.renderbufferStorage( _gl.RENDERBUFFER, glInternalFormat, renderTarget.width, renderTarget.height );

}
```
Are they functional equivalent?

Comment (gkjohnson) [reply src/renderers/webgl/WebGLTextures.js:1446]: Yup! They're the same. I just thought it was easier to read. We can check using a table to make sure they get into the same branches.

*Condition Layout 1*

```js
if ( isUseMultisampledRTT ) {

	// Condition A

} else if ( isMultisample ) {

	// Condition B

}
```

*Table*

| | isUseMultisampledRTT = true | isUseMultisampledRTT = false |
|---|---|---|
| isMultisample = true | A | B |
| isMultisample = false | A | None |

*Condition Layout 2*

```js
if ( isMultisample || isUseMultisampledRTT ) {

	if ( isUseMultisampledRTT ) {

		// Condition A

	} else {

		// Condition B

	}

}
```

*Table*

| | isUseMultisampledRTT = true | isUseMultisampledRTT = false |
|---|---|---|
| isMultisample = true | A | B |
| isMultisample = false | A | None |

*Condition Layout 3*

```js
if ( isMultisample && isUseMultisampledRTT === false ) {

	// Condition B

} else if ( isUseMultisampledRTT ) {

	// Condition A

}
```

*Table*

| | isUseMultisampledRTT = true | isUseMultisampledRTT = false |
|---|---|---|
| isMultisample = true | A | B |
| isMultisample = false | A | None |

Comment (Mugen87) [reply src/renderers/webgl/WebGLTextures.js:1446]: Good! I definitely prefer the PRs approach.

**Review by Mugen87 — APPROVED** [review]

**Pull request merged**